torch/csrc/jit/serialization/pickler.h

```diff
@@ -296,5 +296,32 @@ uint64_t getStorageKey(const at::Tensor& tensor);
 // otherwise return false
 bool checkHasValidSetGetState(const std::shared_ptr<c10::ClassType>& cls);
 
+enum MathBits : int8_t { kCONJ = 1, kNEG };
+
+// return a map of MathBits on Tensor
+inline std::unordered_map<int8_t, bool> getTensorMathBits(const at::Tensor& t) {
+  return {{MathBits::kCONJ, t.is_conj()}, {MathBits::kNEG, t.is_neg()}};
+}
+
+// set MathBits on Tensor from map
+inline void setTensorMathBits(
+    const at::Tensor& t,
+    std::unordered_map<int8_t, bool> math_bits) {
+  t._set_conj(math_bits[MathBits::kCONJ]);
+  t._set_neg(math_bits[MathBits::kNEG]);
+}
+
+// set MathBits on Tensor from map
+inline void setTensorMathBits(
+    const at::Tensor& t,
+    c10::Dict<c10::IValue, c10::IValue> math_bits_idict) {
```

Comment:
This overload is required for C++ `unpickler.cpp` which returns a `c10::Dict<IValue, IValue>`.